pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -207,6 +208,82 @@ func TestUserSignupFailedMissingNSTemplateTier(t *testing.T) {
 		})
 }
 
+func TestUserSignupFailedNoClusterReady(t *testing.T) {
+	// given
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: operatorNamespace,
+			UID:       types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.UserSignupSpec{
+			Username: "foo@redhat.com",
+			Approved: false,
+		},
+	}
+	r, req, _ := prepareReconcile(t, userSignup.Name, userSignup, configMap(config.UserApprovalPolicyAutomatic)) // basicNSTemplateTier does not exist
+	createMemberCluster(r.client, "member1", notReady)
+	defer clearMemberClusters(r.client)
+	// when
+	res, err := r.Reconcile(req)
+	// then
+	// error reported, and request is requeued and userSignup status was updated
+	require.Error(t, err)
+	assert.Equal(t, reconcile.Result{Requeue: false}, res)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+	require.NoError(t, err)
+	t.Logf("usersignup status: %+v", userSignup.Status)
+	test.AssertConditionsMatch(t, userSignup.Status.Conditions,
+		v1alpha1.Condition{
+			Type:   v1alpha1.UserSignupApproved,
+			Status: v1.ConditionTrue,
+			Reason: "ApprovedAutomatically",
+		},
+		v1alpha1.Condition{
+			Type:   v1alpha1.UserSignupComplete,
+			Status: v1.ConditionFalse,
+			Reason: "NoClusterAvailable",
+		})
+}
+
+func TestUserSignupFailedNoClusterWithCapacityAvailable(t *testing.T) {
+	// given
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: operatorNamespace,
+			UID:       types.UID(uuid.NewV4().String()),
```

Comment:
```suggestion
```
same here

Reply:
removed all such `UID` fields in a0a3019